app/controllers/projects_controller.rb

```diff
@@ -23,6 +28,15 @@ def new
 
   # GET /projects/1/edit
   def edit
+    setup_metric_configs(@project)
+    @project.configs.each do |config|
```

Comment:
The loop here checks whether certain credentials has already been given value. In _form.html, it first displays all the existed credentials and adds fields for credentials that haven't been configured.  The cucumber test would pass, and everything still works if we remove this part. But on the edit page, for example, if we already have an url for codeclimate, it still provides an empty field to config url. I hope that makes sense. I should probably modify the cucumber for edit page to test this scenario.